src/Connector.SqlServer/Utils/TableDefinitions/CodeTableDefinition.cs

```diff
@@ -0,0 +1,149 @@
+                    return new ColumnDefinition[]
+                    {
+                        new("EntityId", SqlColumnHelper.UniqueIdentifier, AddIndex: true),
+                        new("Code", SqlColumnHelper.NVarchar1024),
+                        new("ChangeType", SqlColumnHelper.NVarchar256),
+                        new("CorrelationId", SqlColumnHelper.UniqueIdentifier)
+                    };
+
+                case StreamMode.Sync:
+                    return new ColumnDefinition[]
+                    {
+                        new("EntityId", SqlColumnHelper.UniqueIdentifier, AddIndex: true),
+                        new("Code", SqlColumnHelper.NVarchar1024),
+                    };
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(mode), mode, null);
+            }
+        }
+
+        public static IEnumerable<SqlDataRecord> GetSqlRecords(StreamMode mode, SqlConnectorEntityData connectorEntityData)
+        {
+            var columnDefinitions = GetColumnDefinitions(mode);
+
+            switch (mode)
+            {
+                case StreamMode.EventStream:
+                    return connectorEntityData.EntityCodes?.Select(code =>
+                    {
+                        var sqlMetaData = columnDefinitions.Select(column => column.ToSqlMetaData()).ToArray();
+
+                        var record = new SqlDataRecord(sqlMetaData);
+                        record.SetGuid(0, connectorEntityData.EntityId);
+                        record.SetString(1, code.Key);
+                        record.SetString(2, connectorEntityData.ChangeType.ToString());
+                        record.SetGuid(3, connectorEntityData.CorrelationId.Value);
+                        return record;
+                    });
+
+                case StreamMode.Sync:
+                    return connectorEntityData.EntityCodes.Select(code =>
+                    {
+                        var sqlMetaData = columnDefinitions.Select(column => column.ToSqlMetaData()).ToArray();
+
+                        var record = new SqlDataRecord(sqlMetaData);
+                        record.SetGuid(0, connectorEntityData.EntityId);
+                        record.SetString(1, code.Key);
+                        return record;
+                    });
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(mode), mode, null);
+            }
+        }
+
+        public static SqlParameter GetPivotKeyParameter(StreamMode mode, SqlConnectorEntityData connectorEntityData)
+        {
+            return new SqlParameter("@EntityId", SqlDbType.UniqueIdentifier) { Value = connectorEntityData.EntityId };
+        }
+
+        public static SqlServerConnectorCommand CreateUpsertCommand(IReadOnlyStreamModel streamModel, SqlConnectorEntityData connectorEntityData, SqlName schema)
+        {
+            var mode = streamModel.Mode ?? StreamMode.Sync; // TODO: Correct fallback?
```

Comment:
What is the case when it's empty? 
I think we should make the decision and remove TODO. If not us now, who will know better in future? :)